buildpacks/go/faas/middleware_debug.go

```diff
@@ -0,0 +1,33 @@
+// +build debug
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"runtime/debug"
+)
+
+func recoverMiddleware(handler http.Handler) http.Handler {
+	logger := log.New(os.Stdout, "", log.LstdFlags | log.Lmicroseconds)
+	f := func(rw http.ResponseWriter, req *http.Request) {
+		defer func() {
+			if r := recover(); r != nil {
+				recoverError := fmt.Errorf("user function error: %v", r)
+				stack := string(debug.Stack())
+				logger.Printf("%v\n%v\n", recoverError, stack)
+
+				rw.WriteHeader(http.StatusInternalServerError)
+
+				rw.Header().Add("Content-Type", "text/plain")
+				fmt.Fprintln(rw, recoverError)
+				fmt.Fprintln(rw, stack)
```

Comment:
This is only difference between release and debug middleware, only in debug I am sending stacktrace to client. In release only 500 status is returned to client.